state/protocol/protocol_state/protocol_statemachine_test.go

```diff
@@ -79,116 +92,131 @@ func (s *UpdaterSuite) TestTransitionToNextEpochNotAllowed() {
-			entry.InvalidStateTransitionAttempted = true
-		})
-		candidate := unittest.BlockHeaderFixture(
-			unittest.HeaderWithView(protocolState.CurrentEpochSetup.FinalView + 1))
-		updater := newStateMachine(candidate.View, protocolState)
-		err := updater.TransitionToNextEpoch()
-		require.Error(s.T(), err, "should not allow transition to next epoch if we are in Epoch Fallback Mode, "+
-			"i.e. an invalid state transition was attempted")
-	})
 	s.Run("candidate block is not from next epoch", func() {
 		protocolState := unittest.ProtocolStateFixture(unittest.WithNextEpochProtocolState())
 		candidate := unittest.BlockHeaderFixture(
 			unittest.HeaderWithView(protocolState.CurrentEpochSetup.FinalView))
-		updater := newStateMachine(candidate.View, protocolState)
-		err := updater.TransitionToNextEpoch()
-		require.Error(s.T(), err, "should not allow transition to next epoch if we are in Epoch Fallback Mode, "+
-			"i.e. an invalid state transition was attempted")
+		stateMachine, err := newStateMachine(candidate.View, protocolState)
+		require.NoError(s.T(), err)
+		err = stateMachine.TransitionToNextEpoch()
+		require.Error(s.T(), err, "should not allow transition to next epoch if next block is not first block from next epoch")
 	})
 }
 
-// TestBuild tests if the updater returns correct protocol state.
-func (s *UpdaterSuite) TestBuild() {
-	updatedState, stateID, hasChanges := s.updater.Build()
+// TestBuild tests if the protocolStateMachine returns correct protocol state.
+func (s *ProtocolStateMachineSuite) TestBuild() {
+	updatedState, stateID, hasChanges := s.stateMachine.Build()
 	require.Equal(s.T(), stateID, s.parentProtocolState.ID(), "should return same protocol state")
 	require.False(s.T(), hasChanges, "should not have changes")
-	require.NotSame(s.T(), updatedState, s.updater.state, "should return a copy of protocol state")
+	require.NotSame(s.T(), updatedState, s.stateMachine.state, "should return a copy of protocol state")
 	require.Equal(s.T(), updatedState.ID(), stateID, "should return correct ID")
 
-	s.updater.SetInvalidStateTransitionAttempted()
-	updatedState, stateID, hasChanges = s.updater.Build()
-	require.NotEqual(s.T(), stateID, s.parentProtocolState.ID(), "should return same protocol state")
+	updatedDynamicIdentity := s.parentProtocolState.CurrentEpochIdentityTable[0].DynamicIdentity
+	updatedDynamicIdentity.Ejected = true
+	err := s.stateMachine.UpdateIdentity(&flow.DynamicIdentityEntry{
+		NodeID:  s.parentProtocolState.CurrentEpochIdentityTable[0].NodeID,
+		Dynamic: updatedDynamicIdentity,
+	})
+	require.NoError(s.T(), err)
+	updatedState, stateID, hasChanges = s.stateMachine.Build()
 	require.True(s.T(), hasChanges, "should have changes")
+	require.NotEqual(s.T(), stateID, s.parentProtocolState.ID(), "should return same protocol state")
 	require.Equal(s.T(), updatedState.ID(), stateID, "should return correct ID")
 }
```

Comment:
Lets make sure we test 
* parent state has not been changed

```suggestion
	require.Equal(s.T(), s.parentProtocolState.ID(), s.stateMachine.ParentState().ID(), "should not modify parent protocol state")
}
```